src/hotspot/share/oops/methodData.cpp

```diff
@@ -349,7 +349,7 @@ void TypeStackSlotEntries::clean_weak_klass_links(bool always_clean) {
       if (!always_clean && k->is_instance_klass() && InstanceKlass::cast(k)->is_not_initialized()) {
         continue; // skip not-yet-initialized classes // TODO: maybe clear the slot instead?
       }
-      if (always_clean || !k->is_loader_alive() || is_excluded(k)) {
+      if (always_clean || !k->is_loader_present_and_alive() || is_excluded(k)) {
```

Comment:
In the longer term, the thing which concerns me about it is that both `is_excluded()` (and `is_klass_loaded()` in `ciMethodData.cpp`) are deliberately limited to Leyden-related modes while `k->is_loader_present_and_alive()` is performed unconditionally across all execution modes. Seeing a Klass with a missing CLD during normal execution mode is a sign of a bug in some other place.